presto-main/src/test/java/com/facebook/presto/server/TestServer.java

```diff
@@ -229,4 +256,27 @@ public URI uriFor(String path)
     {
         return HttpUriBuilder.uriBuilderFrom(server.getBaseUrl()).replacePath(path).build();
     }
+    private QueryInfo runQuery(String query)
+            throws Exception
+    {
+        Request request = preparePost()
+                .setUri(uriFor("/v1/statement"))
+                .setHeader(PRESTO_USER, "user")
+                .setHeader(PRESTO_CATALOG, "catalog")
+                .setHeader(PRESTO_SCHEMA, "schema")
+                .setBodyGenerator(createStaticBodyGenerator(query, UTF_8))
+                .build();
+
+        QueryResults queryResults = client.execute(request, createJsonResponseHandler(jsonCodec(QueryResults.class)));
+
+        while (queryResults.getNextUri() != null) {
+            queryResults = client.execute(prepareGet().setUri(queryResults.getNextUri()).build(), createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        }
+
+        QueryInfo queryInfo;
+        do {
+            queryInfo = server.getQueryManager().getQueryInfo(new QueryId(queryResults.getId()));
```

Comment:
Thanks for the review, I am trying to change this, however, after I tried to parse JSON back to QueryInfo, I got a parse error:

```
com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.facebook.presto.spi.connector.ConnectorPartitioningHandle: abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
 at [Source: [B@6cd7f381; line: 1, column: 3721] (through reference chain: com.facebook.presto.execution.QueryInfo["outputStage"]->com.facebook.presto.execution.StageInfo["plan"]->com.facebook.presto.sql.planner.PlanFragment["root"]->com.facebook.presto.sql.planner.plan.OutputNode["source"]->com.facebook.presto.sql.planner.plan.ProjectNode["source"]->com.facebook.presto.sql.planner.plan.ExchangeNode["partitioningScheme"]->com.facebook.presto.sql.planner.PartitioningScheme["partitioning"]->com.facebook.presto.sql.planner.Partitioning["handle"]->com.facebook.presto.sql.planner.PartitioningHandle["connectorHandle"])



```
It seems it failed in parsing:
```
connectorHandle: {
@type: "$remote",
partitioning: "FIXED",
function: "ROUND_ROBIN"
}

```
but connectorHandle(ConnectorPartitioningHandle) is a abstract interface in the SPI. How can I fix this?